### DISCUSSION TITLE
Add a simple animation effect for switching pages

### DISCUSSION THREAD

**Discussion by lonelyicer** [vrc-get/vrc-get#1720] [Ideas]
Like this (this is implemented using framer-motion)
![QQ2024121-22217](https://github.com/user-attachments/assets/f7f9dd60-171a-4abb-8dd8-2bc976dccd1d)
Probably makes the page switch look more comfortable

**Comment by anatawa12** [comment]
Adding animations makes appearance beautiful, but I'm worrying that user must wait for animation ends.
It's only 0.few seconds but may affect very experienced users or just affect experience.
Actually I feel slow for some pages with page change animation.

In addition, I'm worrying about minimum macOS version may be affected by this.
for macOS, WebView are updated only on OS updates, so to support legacy macOS versions, we have not to use new css feature.
(JS are easier to use polyfill when needed but css is much harder)

**Reply by lonelyicer** [reply]
> Adding animations makes appearance beautiful, but I'm worrying that user must wait for animation ends.
> It's only 0.few seconds but may affect very experienced users or just affect experience.
> Actually I feel slow for some pages with page change animation.

How about adding a switch to the animation

> In addition, I'm worrying about minimum macOS version may be affected by this.
> for macOS, WebView are updated only on OS updates, so to support legacy macOS versions, we have not to use new css feature.
> (JS are easier to use polyfill when needed but css is much harder)

I don't own a macOS device, so I don't know much about that part of it

edit: safari 12 is starting to support css animations, so I think we shouldn't encounter serious problems with just simple animations as long as the macos version is greater than 10.14

**Reply by anatawa12** [reply]
umm... switch would be an option but if it's off by default i'm considering this should be implemented in ALCOM.

I want to think merits and demerits for few day before make my final decision.


for macos compatibility, I check needed feature is available on older safari on caniuse.

**Reply by JustBuddy** [reply]
I like the idea, it adds a modern look and feel to it.
I would suggest speeding the animation up, though. They don't need to be long to have the desired effect. Making it 2x faster should do the trick.

**Reply by anatawa12** [reply]
I feel speeding up is good. @lonelyicer could you make 2x faster and add to #1672?
